include/boost/geometry/iterators/detail/segment_iterator/range_segment_iterator.hpp

```diff
@@ -100,23 +100,40 @@ class range_segment_iterator
             Reference
         >
 {
+private:
+    static bool has_less_than_two_elements(Range const& r)
+    {
+        iterator_type first = range_iterator_begin<Range>::apply(r);
+        iterator_type last = range_iterator_end<Range>::apply(r);
+
+        return first == last || ++first == last;
+    }
+
 public:
     typedef typename range_iterator_type<Range>::type iterator_type;
 
     // default constructor
     range_segment_iterator()
-        : m_it()
+        : m_it(), m_has_less_than_two_elements()
```

Comment:
As long as we're in C++11 or a constructor of T in C++98 initializes T to 0. Otherwise, this is not true. This is why someone was forced to add a default ctor to ttmath types in the past.